components/aws/sagemaker/tests/integration_tests/component_tests/test_deploy_component.py

```diff
@@ -0,0 +1,108 @@
+    def np2csv(arr):
+        csv = io.BytesIO()
+        numpy.savetxt(csv, arr, delimiter=",", fmt="%g")
+        return csv.getvalue().decode().rstrip()
+
+    # Run prediction on an image
+    runtime = boto3_session.client("sagemaker-runtime")
+    payload = np2csv(train_set[0][30:31])
+
+    response = runtime.invoke_endpoint(
+        EndpointName=endpoint_name, ContentType="text/csv", Body=payload,
+    )
+    return json.loads(response["Body"].read().decode())
+
+
+@pytest.mark.parametrize("test_file_dir", ["resources/config/kmeans-mnist-endpoint"])
+def test_create_endpoint(
+    kfp_client, experiment_id, boto3_session, sagemaker_client, test_file_dir
+):
+
+    download_dir = utils.mkdir(os.path.join(test_file_dir + "/generated"))
+    test_params = utils.load_params(
+        utils.replace_placeholders(
+            os.path.join(test_file_dir, "config.yaml"),
+            os.path.join(download_dir, "config.yaml"),
+        )
+    )
+
+    # Generate random prefix for model, endpoint config and endpoint name
+    # to avoid errors if resources with same name exists
+    test_params["Arguments"]["model_name"] = test_params["Arguments"][
+        "endpoint_config_name"
+    ] = test_params["Arguments"]["endpoint_name"] = input_endpoint_name = (
+        utils.generate_random_string(5) + "-" + test_params["Arguments"]["model_name"]
+    )
+
+    run_id, status, workflow_json = kfp_client_utils.compile_run_monitor_pipeline(
+        kfp_client,
+        experiment_id,
+        test_params["PipelineDefinition"],
+        test_params["Arguments"],
+        download_dir,
+        test_params["TestName"],
+        test_params["Timeout"],
+    )
+    # input_endpoint_name = "gyyka18935-kmeans-mnist-model"
+    # run_id = "1776159b-aeb1-4dc2-8979-545e33db891b"
+    # workflow_json = kfp_client_utils.get_workflow_json(kfp_client, run_id)
+    outputs = {"sagemaker-deploy-model": ["endpoint_name"]}
+
+    output_files = minio_utils.artifact_download_iterator(
+        workflow_json, outputs, download_dir
+    )
+
+    output_endpoint_name = utils.read_from_file_in_tar(
+        output_files["sagemaker-deploy-model"]["endpoint_name"], "endpoint_name.txt"
+    )
+    print(f"endpoint name: {output_endpoint_name}")
+
+    # Verify output from pipeline is endpoint name
+    assert output_endpoint_name == input_endpoint_name
+
+    # Verify endpoint is running
+    assert (
+        sagemaker_utils.describe_endpoint(sagemaker_client, input_endpoint_name)[
+            "EndpointStatus"
+        ]
+        == "InService"
+    )
+
+    # Validate the model for use by running a prediction
+    result = run_predict_mnist(boto3_session, input_endpoint_name, download_dir)
+    print(f"prediction result: {result}")
+    assert json.dumps(result, sort_keys=True) == json.dumps(
+        test_params["ExpectedPrediction"], sort_keys=True
+    )
+
+    # Clean up
+    sagemaker_utils.delete_endpoint(sagemaker_client, input_endpoint_name)
```

Comment:
If any previous step throws an exception, or an assertion fails?, we aren't going to reach this point. Perhaps wrapping in a try/catch/finally would prevent this?

Reply:
Agree but there must be a more pytest-ish way of doing this

Reply:
good catch, will make the change

I tried looking for pytest-ish way, didnt find anything